internal/slicer/report.go

```diff
@@ -0,0 +1,61 @@
+package slicer
+
+import (
+	"fmt"
+	"github.com/canonical/chisel/internal/fsutil"
+	"github.com/canonical/chisel/internal/setup"
+	"io/fs"
+	"sync"
+)
+
+type FileReport struct {
+	Path    string
+	Mode    fs.FileMode
+	Hash    string
+	Size    uint
+	Mutable bool
+	Slices  map[*setup.Slice]bool
```

Comment:
1. How about `map[string]bool` since slices will have unique names?
2. How about using an array/slice instead of a map?

Reply:
1. That was my first implementation. But then I found that I creating a string from the `setup.Slice` in the code each time. Apart from that using the object instead of the representation (even if it is unique) seems to me like it is more robust. Let me turn the question around, what is the benefit of using strings?

Apart from the above, strings almost always require a heap allocation and will take up more memory than the `setup.Slice`  pointers.

2. I am not sure what is the best way. My hunch was that in the future we might want to check whether a particular slice added a file or not, that is easier to do with a map. I will think about it.

Reply:
The one benefit of having strings as keys would be that you can look up if a slice exists without having to fetch the pointer for a particular slice. So, if you wanted to use it for something else from some other package with just the slice name (as string), you would be able to do that.

Reply:
I see. What if we keep using the slice for now and, if the need arises in the future because we are calling it from a different place that only has the string, we change it.

Reply:
Sounds good.